framework/captcha/Captcha.php

```diff
@@ -155,27 +156,16 @@ protected function getClientOptions()
 
         return $options;
     }
-
+    
     /**
      * Checks if there is graphic extension available to generate CAPTCHA images.
      * This method will check the existence of ImageMagick and GD extensions.
-     * @return string the name of the graphic extension, either "imagick" or "gd".
      * @throws InvalidConfigException if neither ImageMagick nor GD is installed.
      */
-    public static function checkRequirements()
+    private function checkRequirements()
     {
-        if (extension_loaded('imagick')) {
-            $imagickFormats = (new \Imagick())->queryFormats('PNG');
-            if (in_array('PNG', $imagickFormats, true)) {
-                return 'imagick';
-            }
-        }
-        if (extension_loaded('gd')) {
-            $gdInfo = gd_info();
-            if (!empty($gdInfo['FreeType Support'])) {
-                return 'gd';
-            }
-        }
-        throw new InvalidConfigException('Either GD PHP extension with FreeType support or ImageMagick PHP extension with PNG support is required.');
+        /* @var $factory DriverFactory */
+        $factory = Yii::createObject(DriverFactory::className());
+        $factory->make();
```

Comment:
I don't get this: method `checkRequirements()` ends with driver instantiation - why?
For me it is obvious `DriverInterface` should contain `checkRequirements()` method, which will be specific per each driver.